scopesim/effects/spectral_trace_list_utils.py

```diff
@@ -481,48 +481,124 @@ def footprint(self, wave_min=None, wave_max=None, xi_min=None, xi_max=None):
+        c : str, optional
+            Colour, any valid matplotlib colour string. The default is "r".
+        axes : matplotlib axes, optional
+            The axes object to use for the plot. If None (default), a new
+            figure with one axes will be created.
+
+        Returns
+        -------
+        axes : matplotlib axes
+            The axes object containing the plot.
+
+        Other Parameters
+        ----------------
+        plot_footprint : bool, optional
+            Plot a rectangle encompassing all control points, which may be
+            larger than the area actually covered by the trace, if the trace is
+            not exactly perpendicular to the detector. The default is True.
+        plot_wave : bool, optional
+            Annotate the wavelength points. The default is True.
+        plot_ctrlpnts : bool, optional
+            Plot the individual control points as makers. The default is True.
+        plot_outline : bool, optional
+            Plot the smallest tetragon encompassing all control points.
+            The default is False.
+        plot_trace_id : bool, optional
+            Write the trace ID in the middle of the trace.
+            The default is False.
+        """
+        if axes is None:
+            _, axes = plt.subplots()
 
         # Footprint (rectangle enclosing the trace)
         xlim, ylim  = self.footprint(wave_min=wave_min, wave_max=wave_max)
         if xlim is None:
-            return
+            return axes
 
         xlim.append(xlim[0])
         ylim.append(ylim[0])
-        plt.plot(xlim, ylim)
+        if plot_footprint:
+            axes.plot(xlim, ylim)
+
+        # for convenience...
+        xname = self.meta["x_colname"]
+        yname = self.meta["y_colname"]
+        wname = self.meta["wave_colname"]
+        sname = self.meta["s_colname"]
 
         # Control points
-        waves = self.table[self.meta["wave_colname"]]
+        waves = self.table[wname]
         if wave_min is None:
             wave_min = waves.min()
         if wave_max is None:
             wave_max = waves.max()
-
-        mask = (waves >= wave_min) * (waves <= wave_max)
-        if sum(mask) > 2:
-            w = waves[mask]
-
-            x = self.table[self.meta["x_colname"]][mask]
-            y = self.table[self.meta["y_colname"]][mask]
-            plt.plot(x, y, "o", c=c)
-
-            for wave in np.unique(w):
-                xx = x[w==wave]
-                xx.sort()
-                dx = xx[-1] - xx[-2]
-
-                plt.text(x[w==wave].max() + 0.5 * dx,
-                         y[w==wave].mean(),
-                         str(wave), va='center', ha='left')
-
-            plt.gca().set_aspect("equal")
+        xis = self.table[sname]
+        if xi_min is None:
+            xi_min = xis.min()
+        if xi_max is None:
+            xi_max = xis.max()
+
+        mask = ((waves >= wave_min) & (waves <= wave_max)
+                & (xis >= xi_min)& (xis <= xi_max))
```

Comment:
I think we need another space here, don't we? ;-)

Reply:
Your eyes are clearly younger than mine 🙃 
```suggestion
                & (xis >= xi_min) & (xis <= xi_max))
```